### PR TITLE
MHV-45982, 45983 - SM - "Start new message", "Folders" urls

### PR DESCRIPTION
**Pull request by alex-morgun-1**
## Summary

- The URL for "Start a new message" page at present is: `https://staging.va.gov/my-health/secure-messages/compose` but the language "compose" is not used elsewhere in the product. Replacing `/compose` with `/new-message` as per recommendations
- Sub-pages within the "My folders" directory do not display the correct URL slug for the parent page (it is currently singular: `/folder` but should be plural: `/folders`)
- adding trailing slash per Staging Review [feedback](https://dsva.slack.com/archives/C03ECSBGSKX/p1684789456399719)
 
## Related issue(s)

- https://jira.devops.va.gov/browse/MHV-45982
- https://jira.devops.va.gov/browse/MHV-45983
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/59488
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/59489

## Testing done

- unit testing coverage
- SQA Testing

## What areas of the site does it impact?

My Health - Secure Messaging

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Comment by oleksii-morgun-va** [comment]
@mdewey tagging you if you have a moment to run by this.

**Review by vsaleem — APPROVED** [review]

**Review by vsaleem — APPROVED** [review]

File: src/applications/mhv/secure-messaging/containers/MessageDetails.jsx [hunk 1]
```diff
@@ -38,7 +38,7 @@ const MessageDetail = () => {
   useEffect(
     () => {
       if (draftMessage?.messageId && message?.draftDate !== null) {
-        history.push(`/draft/${threadId}`);
+        history.push(`/draft/${threadId}/`);
```
Comment (mdewey) [inline src/applications/mhv/secure-messaging/containers/MessageDetails.jsx:41]: Use the PATHs object here as well

File: src/applications/mhv/secure-messaging/containers/Compose.jsx [hunk 2]
```diff
@@ -44,7 +45,7 @@ const Compose = () => {
   useEffect(
     () => {
       if (draftMessage?.messageId && draftMessage.draftDate === null) {
-        history.push('/inbox');
+        history.push('/inbox/');
```
Comment (mdewey) [inline src/applications/mhv/secure-messaging/containers/Compose.jsx:48]: Use the PATHs object here as well

**Review by mdewey — CHANGES_REQUESTED** [review]
I like the `PATHS` constant. I see a few spots where it was missed. Unless there is a reason not to use in those few places.

**Comment by oleksii-morgun-va** [comment]
> I like the `PATHS` constant. I see a few spots where it was missed. Unless there is a reason not to use in those few places.

That was not intentional. Thanks for spotting 👍🏻

File: src/applications/mhv/secure-messaging/components/shared/SmBreadcrumbs.jsx [hunk 3]
```diff
@@ -21,6 +22,14 @@ const SmBreadcrumbs = () => {
       setIsMobile(false);
     }
   }
+
+  const [, locationBasePath, locationChildPath] = useMemo(
```
[the remark this answers is not in the document]
Comment (oleksii-morgun-va) [reply src/applications/mhv/secure-messaging/components/shared/SmBreadcrumbs.jsx:26]: I see that however this was implement previously, the intent is to omit the first element of the array while destructuring. I was just copying these line from another place to be honest. But I guess I can make it look cleaner

**Review by vsaleem — COMMENTED** [review]
A few minor comments!

**Review by vsaleem — APPROVED** [review]

**Review by vsaleem — APPROVED** [review]

File: src/applications/mhv/secure-messaging/reducers/breadcrumbs.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import { Actions } from '../util/actionTypes';
 
 const initialState = {
-  list: [],
+  list: {},
```
Comment (mdewey) [inline src/applications/mhv/secure-messaging/reducers/breadcrumbs.js:4]: Was this change intentional?

**Review by mdewey — APPROVED** [review]
Looks like there are some breadcrumbs changes. Was supposed to be part of this PR? 

Just wanted make sure something weird isn't going on with git

File: src/applications/mhv/secure-messaging/reducers/breadcrumbs.js (hunk 4 above)
Comment (oleksii-morgun-va) [reply src/applications/mhv/secure-messaging/reducers/breadcrumbs.js:4]: initially, that object was an array. However, there is no need to have an array since we are using only a single level breadcrumb now
